src/librustc_passes/error_codes.rs

```diff
@@ -198,6 +198,30 @@ impl Foo for Bar {
 ```
 "##,
 
+E0561: r##"
+A pattern has been used as an argument of a function pointer types.
+
+Erroneous code example:
+
+```compile_fail,E0561
+type A1 = fn(mut arg: u8); // error!
+type A2 = fn(&arg: u32); // error!
+```
+
+When using an alias over a function type, you can't specify things such as an
+argument being mutable, it wouldn't be able to represent functions matching this
```

Comment:
We could easily allow this but we don't because it is redundant. I'd remove "it wouldn't..."